ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1337 @@
+
+                tel = TelescopeDescription.guess(*pix_pos,
+                                                 equivalent_focal_length=foclen)
+                tel.optics.mirror_area = mirror_area
+                tel.optics.num_mirror_tiles = num_tiles
+                subarray.tels[tel_id] = tel
+                subarray.positions[tel_id] = tel_pos
+
+            except self.pyhessio.HessioGeneralError:
+                pass
+
+        return subarray
+
+
+class MAGICEventSourceROOT(EventSource):
+    """
+    EventSource for MAGIC calibrated data.
+
+    This class operates with the MAGIC data run-wise. This means that the files
+    corresponding to the same data run are loaded and processed together.
+    """
+    _count = 0
+
+    def __init__(self, config=None, tool=None, **kwargs):
+        """
+        Constructor
+
+        Parameters
+        ----------
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        tool: ctapipe.core.Tool
+            Tool executable that is calling this component.
+            Passes the correct logger to the component.
+            Set to None if no Tool to pass.
+        kwargs: dict
+            Additional parameters to be passed.
+            NOTE: The file mask of the data to read can be passed with
+            the 'input_url' parameter.
+        """
+
+        file_list = glob.glob(kwargs['input_url'])
+        file_list.sort()
+
+        # EventSource can not handle file wild cards as input_url
+        # To overcome this we substitute the input_url with first file matching
+        # the specified file mask.
+        del kwargs['input_url']
+        super().__init__(config=config, tool=tool, input_url=file_list[0], **kwargs)
+
+        # Retrieving the list of run numbers corresponding to the data files
+        run_numbers = list(map(self._get_run_number, file_list))
+        self.run_numbers = np.unique(run_numbers)
+
+        # # Setting up the current run with the first run present in the data
+        # self.current_run = self._set_active_run(run_number=0)
+        self.current_run = None
+        
+        # MAGIC telescope positions in m wrt. to the center of CTA simulations
+        self.magic_tel_positions = {
+            1: [-27.24, -146.66, 50.00] * u.m,
+            2: [-96.44, -96.77, 51.00] * u.m
+        }
+        # MAGIC telescope description
+        optics = OpticsDescription.from_name('MAGIC')
+        geom = CameraGeometry.from_name('MAGICCam')
+        self.magic_tel_description = TelescopeDescription(optics=optics, camera=geom)
+        self.magic_tel_descriptions = {1: self.magic_tel_description, 2: self.magic_tel_description}
+        self.magic_subarray = SubarrayDescription('MAGIC', self.magic_tel_positions, self.magic_tel_descriptions)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        return
```

Comment:
an empty  `__exit__`  is inherited from `EventSource` no need to explicitly write it down

Reply:
Removed...